### PR TITLE
Issue #925: Added noncopy constructor for Tensor and TensorT

### PR DESCRIPTION
**Pull request by ThePseudo**
Added methods reserve() and constructor for Tensor and TensorT

### CODE REVIEW

**Comment by axsaucedo** [comment]
Looks good - only thing would be to add a test, and we can then merge. You can add this on the Tensor test file.

**Comment by axsaucedo** [comment]
Just following up, do you think you would be able to have a look with a short test @ThePseudo  ?

**Comment by ThePseudo** [comment]
Sorry for the delay, at some point stuff happened and I lost track of a few things (one being this). I created a short test, which could be expanded upon, but is basically "inspired" by the other test already present. I hope this works!

**Comment by axsaucedo** [comment]
No worries @ThePseudo, thank you for following up - running test suite

**Comment by axsaucedo** [comment]
@ThePseudo it seems tests failing, did you run locally?

**Comment by ThePseudo** [comment]
The terrible thing is that I am not really an expert in the Google test suite... but I found that the tests were failing for a good reason, apparently, and it is building. Which is strange, since "it worked on my machine", but let's try this out

File: src/include/kompute/Tensor.hpp [hunk 1]
```diff
@@ -79,6 +96,13 @@ class Tensor
                  uint32_t elementTotalCount,
                  uint32_t elementMemorySize);
 
+    /**
+     * @brief Reserve memory on the tensor
+     *
+     * @param newSize the new size for reservation
```
Comment (axsaucedo) [inline src/include/kompute/Tensor.hpp:102]: Docstrings need update as well

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Thank you @ThePseudo - remaining commment on a small docstring change required

**Comment by ThePseudo** [comment]
> Thank you @ThePseudo - remaining commment on a small docstring change required

It should be done now, but tell me if the format is not correct (I took inspiration from previous docstrings)

File: src/include/kompute/Tensor.hpp (hunk 1 above)
Comment (axsaucedo) [reply src/include/kompute/Tensor.hpp:103]:
```suggestion
     * @param elementTotalCount Total number of elements for new size
     * @param elementMemorySize Memory size of element 
```

It has to match the number of parameters, see suggestion

**Comment by axsaucedo** [comment]
Added extra suggestion as it needs to follow the same format as rest of docstrings

**Comment by axsaucedo** [comment]
Thank you @ThePseudo - signed commit required, and we should be able to wrap it up :)

**Comment by ThePseudo** [comment]
> Thank you @ThePseudo - signed commit required, and we should be able to wrap it up :)

Forgot it, now added! :D

**Comment by axsaucedo** [comment]
Thanks!

**Pull request merged**